crypto/secp256k1/secp256.go

```diff
@@ -49,6 +49,41 @@ var (
 	ErrRecoverFailed       = errors.New("recovery failed")
 )
 
+func VerifySeckeyValidity(seckey []byte) error {
+	if len(seckey) != 32 {
+		return errors.New("priv key is not 32 bytes")
+	}
+	var seckey_ptr *C.uchar = (*C.uchar)(unsafe.Pointer(&seckey[0]))
+	ret := C.secp256k1_ec_seckey_verify(context, seckey_ptr)
+	if int(ret) != 1 {
+		return errors.New("invalid seckey")
+	}
+	return nil
+}
+
+func GeneratePubKey(seckey []byte) ([]byte, error) {
```

Comment:
In order to comply with the existing naming convention(e.g. function name DecompressPubkey), maybe `GeneratePubKey` can be modified to` GeneratePubkey`.

By the way, is this new function triggered by any use case, or is it implemented for future usage?